File: Linked List/Linked List template.c
#include<stdio.h>
#include<stdlib.h>
struct node {
    int data;
    struct node *next;
};
void insert_at_back(struct node **head, int data) {
    if(*head == NULL) {
        (*head) = (struct node *)malloc(sizeof(struct node));
        (*head)->data=data;
        (*head)->next=NULL;
        return;
    }
    struct node *tmp = *head;
    while(tmp->next != NULL) {
        tmp = tmp->next;
    }
    tmp->next=(struct node *)malloc(sizeof(struct node));
    tmp=tmp->next;
    tmp->data=data;
    tmp->next=NULL;
}
void insert_at_front(struct node **head,int data) {
    if(*head == NULL) {
        (*head) = (struct node *)malloc(sizeof(struct node));
        (*head)->data=data;
        (*head)->next=NULL;
        return;
    }
    struct node *tmp = (struct node *)malloc(sizeof(struct node));
    tmp->data=data;
    tmp->next=(*head);
    (*head)=tmp;
}
void delete_last(struct node **head) {
    if(*head == NULL) {
        return;
    }
    struct node *last = *head;
    struct node *prev_last;
    while( last->next != NULL ) {
        prev_last = last;
        last = last->next;
    }
    prev_last->next=NULL;
    free(last);
}
void delete_front(struct node **head) {
    if(*head == NULL) {
        return;
    }
    struct node *tmp = *head;
    if(tmp->next == NULL) {
        *head=NULL;
        free(tmp);
        return;
    }
    (*head) = tmp->next;
    free(tmp);
}
void display(struct node **head) {
    if(*head == NULL) {
        printf("list is empty\n");
        return;
    }
    struct node *tmp = *head;
    while(tmp->next != NULL) {
        printf("%d ",tmp->data);
        tmp=tmp->next;
    }
    printf("%d\n",tmp->data);
}
int main() {

    struct node *head = NULL;

    insert_at_back(&head, 23);
    insert_at_back(&head, 28);
    insert_at_back(&head, 28);
    insert_at_back(&head, 35);
    insert_at_back(&head, 49);
    insert_at_back(&head, 49);
    insert_at_back(&head, 53);
    insert_at_back(&head, 53);
    insert_at_back(&head, 17);
    insert_at_back(&head, 18);
    insert_at_back(&head, 18);
    insert_at_back(&head, 44);

    display(&head);
    delete_front(&head);
    display(&head);

    delete_last(&head);
    display(&head);

    insert_at_front(&head, 222);
    display(&head);

    return 0;
}

File: Linked List/LinkedList_template.java
public class LinkedList_template {
    public static class LinkedList {
        static node head;
        public static class node {
            int data;
            node next;
            node(int data) {
                this.data=data;
                this.next=null;
            }
        }
        public static void insert_at_back(int data){
            if(head == null) {
                head = new node(data);
                return;
            }
            node tmp = head;
            while(tmp.next != null) {
                tmp=tmp.next;
            }
            tmp.next = new node(data);
            tmp=tmp.next;
            tmp.next=null;
            return;
        }
        public static void insert_at_front(int data) {
            if(head == null) {
                head = new node(data);
                return;
            }
            node tmp = new node(data);
            tmp.next=head;
            head=tmp;
            return;
        }
        public static void delete_front() {
            if(head == null) {
                return;
            }
            node tmp=head;
            if(tmp.next == null) {
                head = null;
                return;
            }
            head=tmp.next;
        }
        public static void delete_back() {
            if(head == null) {
                System.out.println("list is empty");
                return;
            }
            node tmp = head;
            node prev = head;
            while(tmp.next != null) {
                prev = tmp;
                tmp = tmp.next;
            }
            prev.next=null;
        }
        public static void display() {
            if(head == null) {
                System.out.println("list is empty");
                return;
            }
            node tmp = head;
            while(tmp != null) {
                System.out.print(tmp.data+" ");
                tmp = tmp.next;
            }
            System.out.println();
        }
    }
    public static void main(String args[]) {
        LinkedList list = new LinkedList();

        list.insert_at_back(11);
        list.insert_at_back(12);
        list.insert_at_back(13);
        list.insert_at_back(14);
        list.insert_at_back(15);
        list.insert_at_back(56);

        list.insert_at_front(99);
        list.insert_at_front(89);
        list.insert_at_front(67);

        list.display();

        list.delete_front();
        list.display();

        list.delete_back();
        list.display();
    }
}
